mailbox.go

```diff
@@ -18,28 +18,85 @@ var (
 	errDeregisteredFailed = errors.New("grid: deregistered failed")
 )
 
+// mailboxRegistry is a collection of named mailboxes.
+type mailboxRegistry struct {
+	mu sync.RWMutex
+	r  map[string]*Mailbox
+}
+
+func newMailboxRegistry() *mailboxRegistry {
+	return &mailboxRegistry{
+		r: make(map[string]*Mailbox),
+	}
+}
+
+// Get retrieves the mailbox.
+func (r *mailboxRegistry) Get(name string) (m *Mailbox, found bool) {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	m, found = r.r[name]
+	return
+}
+
+// Set the mailbox.
+func (r *mailboxRegistry) Set(name string, m *Mailbox) (update bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	_, update = r.r[name]
+	r.r[name] = m
+	return
+}
+
+// Delete the mailbox.
+func (r *mailboxRegistry) Delete(name string) (found bool) {
+	r.mu.Lock()
+	defer r.mu.Unlock()
+	_, found = r.r[name]
+	if found {
+		delete(r.r, name)
+	}
+	return
+}
+
+// Size of the registry.
+func (r *mailboxRegistry) Size() int {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return len(r.r)
+}
+
+// R returns the underlying registry.
+// It does _not_ return a copy.
```

Comment:
What do you mean?  It looks like it DOES return a copy of the mailbox map... 

I assume this is for testing only, so I'd add that comment.

Reply:
@epsniff Modifying the map returned here modifying the map stored on the struct: https://go.dev/play/p/gBap9sul28C. That's what I meant by not a copy: if the returned map were a copy you could change it without impacting the struct's map.

Reply:
I'd just go ahead and produce a copy of the map and make the func safe.   There isn't a performance reason to not copy it.  This function is called enough to try doing anything fancy.   It's better to no have a future foot gun.